### PR TITLE
Audit module dependencies

### PR DESCRIPTION
**Pull request by Pessimistress**
For #7457 

#### Change List
- Remove duplicates between dependencies and peerDependencies
- peerDependencies should not lock version
- Remove unused dependencies

### CODE REVIEW

File: modules/jupyter-widget/package.json [hunk 1]
```diff
@@ -40,6 +40,7 @@
     "@loaders.gl/core": "^4.1.0",
     "@loaders.gl/csv": "^4.1.0",
     "@luma.gl/core": "9.0.0-beta.4",
+    "d3-dsv": "^1.0.8",
```
Comment (Pessimistress) [inline modules/jupyter-widget/package.json:43]: `@loaders.gl/csv` is missing this dependency @ibgreen

File: modules/aggregation-layers/package.json [hunk 2]
```diff
@@ -38,18 +38,16 @@
...
     "@luma.gl/shadertools": "9.0.0-beta.4",
     "@math.gl/web-mercator": "^4.0.0",
     "d3-hexbin": "^0.2.1"
   },
   "peerDependencies": {
     "@deck.gl/core": "^9.0.0-alpha",
     "@deck.gl/layers": "^9.0.0-alpha",
-    "@luma.gl/core": "9.0.0-beta.4",
-    "@luma.gl/engine": "9.0.0-beta.4"
+    "@luma.gl/core": "^9.0.0-beta",
```
Comment (ibgreen) [inline modules/aggregation-layers/package.json:49]: Users don't have to explicitly specify luma deps right? they are covered by deck.gl/core or something? Why is shadertools handled differently?

File: modules/core/src/lib/attribute/data-column.ts [hunk 3]
```diff
@@ -509,20 +520,20 @@ export default class DataColumn<Options, State> {
   // https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/vertexAttribPointer
   normalizeConstant(value: NumericArray): NumericArray {
```
Comment (ibgreen) [inline modules/core/src/lib/attribute/data-column.ts:521]: Should this always return a `Float32Array`?

If we are using luma.gl types, should this be a luma.gl utility?

**Review by ibgreen — APPROVED** [review]

File: modules/aggregation-layers/package.json (hunk 2 above)
Comment (Pessimistress) [reply modules/aggregation-layers/package.json:49]: Technically the only package that has to go into peer dep is core, which does not allow multiple copies.

**Pull request merged**